client/locations-list-map.js

```diff
@@ -171,293 +187,511 @@ function updateFilters(filters) {
+  // Text to go into InfoWindow
+  const contentTags = separator ? [ce('h5', 'separator', ctn(name))] : [ce('h5', null, ctn(name))];
 
-        if (filterOrgTypeKeys) {
-          const acc = (entry.org_type || "").toLowerCase();
-          if (!filterOrgTypeKeys.some(s => acc === s)) {
-            return;
-          }
-        }
+  if (orgType && orgType.length) {
+    contentTags.push(
+      ce('div', 'label', ctn($.i18n('ftm-maps-marker-org-type-label'))),
+      ce('div', 'value', ctn(translateEnumValue(orgType))),
+    );
+  }
 
-        if (entry.marker) {
-          const mapBounds = map.getBounds();
+  if (address) {
+    contentTags.push(
+      ce('div', 'label', ctn($.i18n('ftm-maps-marker-address-label'))),
+      ce('div', 'value', createMapLink(address)),
+    );
+  }
 
-          if (mapBounds && !mapBounds.contains(entry.marker.getPosition())) {
-            return;
-          }
-        }
+  if (instructions) {
+    contentTags.push(
+      ce('div', 'label', ctn($.i18n('ftm-maps-marker-instructions-label'))),
+      linkifyElement(ce('div', 'value', multilineStringToNodes(instructions))),
+    );
+  }
 
-        listCount++;
-        entry.cityName = cityName;
-        entry.stateName = stateName;
-        entries.push(entry);
-      });
-    }
+  if (accepting) {
+    contentTags.push(
+      ce('div', 'label', ctn($.i18n('ftm-maps-marker-accepting-label'))),
+      ce('div', 'value', ctn(translateEnumList(accepting))),
+    );
   }
 
-  // TODO: This is hacky since technically this function should ONLY be responsible for generating HTML snippets,
-  //  not updating stats; however this is the quickest method for updating filter stats as well.
-  if (map) {
-    // if the map hasn't loaded yet, don't update requester count - otherwise it'll flash once the map uploads (depending on zoom level)
-    updateStats($('#list-stats'), listCount);
+  if (openBox) {
+    contentTags.push(
+      ce('div', 'label', ctn($.i18n('ftm-maps-marker-open-packages-label'))),
+      ce('div', 'value', ctn(translateEnumValue(openBox))),
+    );
   }
 
-  return entries;
+  return contentTags;
 }
 
-function getDataJsonUrl(filename) {
-  return `/${filename}`
-}
+function createMarker(latitude, longitude, orgType, address, name, instructions, accepting,
+  openBox, markerOptions, otherRequesters) {
+  const location = { lat: latitude, lng: longitude };
+  const options = {
+    position: location,
+    title: name,
+    ...markerOptions || {},
+  };
+  const marker = new google.maps.Marker(options);
 
-function getCountryDataFilename(country) {
-  // Always use country-specific data.json file
-  let countryDataFilename;
+  marker.addListener('click', () => {
+    sendEvent('map', 'click', 'marker');
 
-  countryDataFilename = `data-${country}.json`;
-  return countryDataFilename;
-}
+    gOpenInfoWindows.forEach((infowindow) => infowindow.close());
+    gOpenInfoWindows = [];
```

Comment:
Could make gOpenInfoWindows const, and set its length to 0 to clear it. Reassigning the global seems weird.